### PR TITLE
Enforce security headers in Fleet Manager API

### PR DESCRIPTION
**Pull request by kurlov**
## Description

Add recommended security headers to Fleet Manger envoy config
Most important:
- HTTP Strict Transport Security (HSTS)
- Content-Security-Policy

See: https://wiki.owasp.org/index.php/OWASP_Secure_Headers_Project#tab=Headers 

## Checklist (Definition of Done)

~~- [ ] Unit and integration tests added~~
~~- [ ] Added test description under `Test manual`~~
~~- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~~
- [x] CI and all relevant tests are passing
~~- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`~~
~~- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~~
~~- [ ] Add secret to app-interface Vault or Secrets Manager if necessary~~
~~- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)~~
~~- [ ] Check AWS limits are reasonable for changes provisioning new resources~~

## Test manual

Only on stage env

### CODE REVIEW

File: templates/service-template.yml [hunk 1]
```diff
@@ -860,6 +860,27 @@ objects:
...
+                        route:
+                          cluster: backend
+                        typed_per_filter_config:
+                          lua_security_headers:
+                            "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.LuaPerRoute
+                            source_code:
+                              inline_string: |
+                                function envoy_on_response(response_handle)
+                                  csp = "default-src 'none'; script-src 'self'; img-src 'self'; style-src 'self'; connect-src 'self'";
+                                  response_handle:headers():add("Content-Security-Policy", csp);
```
Comment (SimonBaeumer) [inline templates/service-template.yml:875]: What does `csp` mean in this context?

File: templates/service-template.yml [hunk 2]
```diff
@@ -860,6 +860,27 @@ objects:
...
+                                  response_handle:headers():add("X-Frame-Options", "deny");
+                                  response_handle:headers():add("X-XSS-Protection", "1; mode=block");
+                                  response_handle:headers():add("X-Content-Type-Options", "nosniff");
+                                  response_handle:headers():add("Referrer-Policy", "no-referrer");
+                                  response_handle:headers():add("X-Download-Options", "noopen");
+                                  response_handle:headers():add("X-DNS-Prefetch-Control", "off");
+                                  response_handle:headers():add("Strict-Transport-Security", "max-age=31536000; includeSubDomains");
```
Comment (SimonBaeumer) [inline templates/service-template.yml:882]: Recommendation according to [mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#examples).
`Strict-Transport-Security: max-age=63072000; includeSubDomains; preload`

**Review by SimonBaeumer — CHANGES_REQUESTED** [review]

File: templates/service-template.yml (hunk 1 above)
Comment (kurlov) [reply templates/service-template.yml:875]: It stands for Content Security Policy

Comment (SimonBaeumer) [reply templates/service-template.yml:875]: Ah, I see it is a variable. Overlooked it.

**Review by SimonBaeumer — APPROVED** [review]

**Review by vjwilson — APPROVED** [review]

**Pull request merged**